streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.streams.sprinklr.util;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.apache.commons.lang.StringUtils;
+import org.apache.streams.pojo.extensions.ExtensionUtil;
+import org.apache.streams.pojo.json.*;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.List;
+
+import static org.apache.streams.data.util.ActivityUtil.*;
+
+/**
+ * Converts SprinklrData to an Activity Object
+ */
+public class SprinklrDataToActivityConverter {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SprinklrDataToActivityConverter.class);
+    public static final String LANGUAGE_KEY = "LANGUAGE";
+
+    /**
+     * convert the given Sprinklr data in JsonNode format to an Activity
+     * @param item JsonNode
+     * @return an Activity
+     */
+    public Activity convert(JsonNode item) {
+        Activity activity = new Activity();
+
+        try {
+            activity.setVerb("post");
+            activity.setId(formatId(activity.getVerb(), item.get("universalMessageId").asText()));
+            activity.setContent(item.has("message") ? item.get("message").asText() : "");
+            activity.setTitle(item.has("title") ? item.get("title").asText() : "");
+            activity.setPublished(item.has("createdTime") ? new DateTime(item.get("createdTime").asLong()) : new DateTime());
```

Comment:
I went with the content creation time.

Per the documentation:

- ‘SYSTEM_CREATED_TIME’ - system created time. 
- ‘CREATED_TIME’ - social network created time. 
- ‘MODIFIED_TIME’ - modified time.